AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/smithygenerated/aws_cryptography_keystore/client.py

```diff
@@ -0,0 +1,462 @@
+            for interceptor in interceptors:
+                context._request = interceptor.modify_before_serialization(context)
+
+            # Step 3: Invoke the read_before_serialization hooks
+            for interceptor in interceptors:
+                interceptor.read_before_serialization(context)
+
+            # Step 4: Serialize the request
+            context_with_transport_request = cast(
+                InterceptorContext[Input, None, DafnyRequest, None], context
+            )
+            context_with_transport_request._transport_request = await serialize(
+                context_with_transport_request.request, config
+            )
+
+            # Step 5: Invoke read_after_serialization
+            for interceptor in interceptors:
+                interceptor.read_after_serialization(context_with_transport_request)
+
+            # Step 6: Invoke modify_before_retry_loop
+            for interceptor in interceptors:
+                context_with_transport_request._transport_request = (
+                    interceptor.modify_before_retry_loop(context_with_transport_request)
+                )
+
+            # Step 7: Acquire the retry token.
+            retry_strategy = config.retry_strategy
+            retry_token = retry_strategy.acquire_initial_retry_token()
+
+            while True:
+                # Make an attempt, creating a copy of the context so we don't pass
+                # around old data.
+                context_with_response = await self._handle_attempt(
+                    deserialize,
+                    interceptors,
+                    context_with_transport_request.copy(),
+                    config,
+                    operation_name,
+                )
+
+                # We perform this type-ignored re-assignment because `context` needs
+                # to point at the latest context so it can be generically handled
+                # later on. This is only an issue here because we've created a copy,
+                # so we're no longer simply pointing at the same object in memory
+                # with different names and type hints. It is possible to address this
+                # without having to fall back to the type ignore, but it would impose
+                # unnecessary runtime costs.
+                context = context_with_response  # type: ignore
+
+                if isinstance(context_with_response.response, Exception):
+                    # Step 7u: Reacquire retry token if the attempt failed
+                    try:
+                        retry_token = retry_strategy.refresh_retry_token_for_retry(
+                            token_to_renew=retry_token,
+                            error_info=RetryErrorInfo(
+                                # TODO: Determine the error type.
+                                error_type=RetryErrorType.CLIENT_ERROR,
+                            ),
+                        )
```

Comment:
We shouldn't retry here. It's a local call and not a service call so it's highly unlikely that an error is a transient one.

Reply:
Correct, we aren't retrying. This structure comes from Smithy-Python. I've generated a no-retry, no-delay token for use here: https://github.com/aws/aws-cryptographic-material-providers-library/pull/225/files#diff-cdf1f2f63adc592b23747994bd05ead4ec705088b8b4d0f36df2808485d073e1